src/dataframe/dataframe.jl

```diff
@@ -182,14 +221,15 @@ function DataFrame(pairs::Pair{<:AbstractString,<:Any}...; makeunique::Bool=fals
                      copycols=copycols)
 end
 
-# these two are needed as a workaround Tables.jl dispatch
-DataFrame(pairs::AbstractVector{<:Pair}; makeunique::Bool=false,
-          copycols::Bool=true) =
-    DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
-
-DataFrame(pairs::NTuple{N, Pair}; makeunique::Bool=false,
-          copycols::Bool=true) where {N} =
-    DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
+# this is needed as a workaround for Tables.jl dispatch
+function DataFrame(pairs::AbstractVector{<:Pair}; makeunique::Bool=false,
+                   copycols::Bool=true)
+    if isempty(pairs)
+        return DataFrame()
+    else
+        return DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
+    end
+end
```

Comment:
no, because `DataFrame(makeunique=makeunique, copycols=copycols)` as it will dispatch to kwarg constructor in which  `makeunique` or `copycols` keyword arguments are not allowed. I will fix the splatting.

Reply:
fixed the constructor.